picard/util/__init__.py

```diff
@@ -1183,4 +1190,14 @@ def detect_unicode_encoding(path):
         for bom, encoding in ENCODING_BOMS.items():
             if first_bytes.startswith(bom):
                 return encoding
-        return 'utf-8'
+
+        if detect is None:
+            return 'utf-8'
+
+        rawdata = f.read(read_limit)
+        result = detect(first_bytes + rawdata)
```

Comment:
Instead of creating a new string by concatenation, we should just use `seek()`:

```python
f.seek(0)
result = detect(f.read(read_limit))
```

`rawdata` intermediary variable isn't strictly needed either.